azure-pipelines.yaml

```diff
@@ -47,10 +47,10 @@ steps:
   displayName: set safe container tag as lower case container tag with illegal characters removed
   # extracts merged PR number from git log if a PR merge to master
 - script: |
-    gitLog=`git log -1 --pretty=%B`
-    mergedPrNo=`echo $gitLog | sed -n 's/Merge pull request #\(.*\) from .*/\1/p'`
+    gitLog=`git log --pretty=oneline --abbrev-commit -1`
+    mergedPrNo=`echo $gitLog | sed -n 's/.*(#\([0-9]\+\)).*/\1/p'`
     echo '##vso[task.setvariable variable=mergedPrNo]'$mergedPrNo
-  condition: and(succeeded(), contains(variables['Build.SourceBranch'], 'refs/heads/$(Build.SourceBranchName)'))
+  condition: and(succeeded(), contains(variables['Build.SourceBranch'], 'refs/heads/master'))
```

Comment:
The Build.SourceBranchName wasn't working as expected previously and was skipping the step during a merge.

However it may have been due to a syntax error in the 'condition' script. 
I've changed the syntax of the line to be:

`condition: and(succeeded(), contains(variables['Build.SourceBranch'], format('refs/heads/{0}', variables['Build.SourceBranchName'])))`

This works to exclude the step when not a merge, but we can only test the merge works during a merge. 

Suggest merging this, and if it works all is good, otherwise create a new PR to reinstate the 'master' check.

But hopefully it will work.

Reply:
Makes sense.